Blockzilla/BrowserViewController.swift

```diff
@@ -215,11 +219,50 @@ class BrowserViewController: UIViewController {
         super.viewDidAppear(animated)
     }
     
+    private func setupBiometrics() {
+        // TODO: Localize this
+        self.context.localizedReason = "Authenticate to continue session."
+        self.context.localizedCancelTitle = "New session"
+        
+        // Register for foreground notification to check biometric authentication
+        NotificationCenter.default.addObserver(forName: NSNotification.Name.UIApplicationWillEnterForeground, object: nil, queue: nil) { notification in
+            var biometricError: NSError?
+
+            // Check if user is already in a cleared session
+            if !self.webViewContainer.isHidden {
+                if (Settings.getToggle(SettingsToggle.biometricLogin)) {
+                    self.displaySplashScreen()
+                    
+                    if self.context.canEvaluatePolicy(LAPolicy.deviceOwnerAuthenticationWithBiometrics, error: &biometricError) {
+                        self.context.evaluatePolicy(.deviceOwnerAuthenticationWithBiometrics, localizedReason: self.context.localizedReason) {
+                            [unowned self] (success, authenticationError) in
+                            
+                            DispatchQueue.main.async {
+                                if success {
+                                    self.hideSplashScreen()
+                                    self.showToolbars()
+                                } else {
+                                    // Clear the browser session, as the user failed to authenticate.
+                                    self.resetBrowser()
+                                    self.hideSplashScreen()
```

Comment:
does this need to be in a specific order, otherwise don't repeat this line in both if-conditions